src/main/java/com/google/ehub/data/CommentDataManager.java

```diff
@@ -54,7 +57,8 @@ public void addItemComment(long itemId, String comment, long timestampMillis) {
     for (Entity entity : comments.asIterable()) {
       String comment = (String) entity.getProperty(COMMENT_PROPERTY_KEY);
       long timestampMillis = (Long) entity.getProperty(TIMESTAMP_PROPERTY_KEY);
-      results.add(new CommentData(itemId, comment, timestampMillis));
+      String username = (String) entity.getProperty(USERNAME_PROPERTY_KEY);
```

Comment:
If you store the user email in the datastore (as suggested in the comment below), you can convert the user email to username here.